apps/prairielearn/src/lib/timezones.ts

```diff
@@ -20,7 +20,23 @@ async function getAvailableTimezonesFromDB(): Promise<Timezone[]> {
 
 export async function getAvailableTimezones(): Promise<Timezone[]> {
   if (memoizedAvailableTimezones == null) {
-    memoizedAvailableTimezones = await getAvailableTimezonesFromDB();
+    // Different portions of the code use the timezone in either PostgreSQL or
+    // Intl, so we return only those timezones that are supported by both. While
+    // Intl provides a list of supported timezones via Intl.supportedValuesOf(),
+    // the list is not exhaustive and may not include some timezones supported
+    // by JS as aliases. Instead, we attempt to create a new DateTimeFormat
```

Comment:
To clarify about this limitation: a first version of this change filtered by `supportedValuesOf()`, which caused a large list of timezones that [are aliases to other timezones](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Temporal/ZonedDateTime#time_zones_and_offsets) not to be included. For example, some timezones like `MST` or `US/Arizona` are actually aliases to `America/Phoenix`, and are supported by `Intl.DateTimeFormat()`, but are not listed in `supportedValuesOf()`. I believe that it makes sense to give instructors access to these aliases, though I see the value of simplifying the dropdown list to show a shorter (but still comprehensive) list. I'd welcome any thoughts.

Reply:
I do like the idea of having a shorter list if possible -- it's very long at the moment. But I haven't had a chance to look at the `supportedValuesOf()` list to see whether it's missing important timezones.

Reply:
Oh cool. Looking through that list I think it would be great to only use the Canonical names. This both keeps the list shorter and removes some of the more confusing things (like "EST" which is a link to "America/Panama", which is not what anyone would want). I think the non-canonical names are mainly there for backward compatibility, which we don't care about in this context.

Reply:
Any chance of checking what users are using, to double check that nobody is using a linked timezone? This wouldn't be a problem in most cases, but if the dropdown does not list the timezone the have currently on it may cause confusion in the Settings page.

```sql
SELECT DISTINCT display_timezone FROM pl_courses UNION
SELECT DISTINCT display_timezone FROM course_instances UNION
SELECT DISTINCT display_timezone FROM institutions;
```

Reply:
One option is to have two lists of timezones: accepted timezones (all including links) and listed timezones (only canonical). The dropdown could list only canonical entries, but if the current value for the course/course instance is not canonical it could include the currently set timezone with appropriate values. In other words, as long as the timezone the current course has is valid, the dropdown would list it in conjunction with canonical timezones, but as soon as the instructor changes it to a canonical entry, it would no longer be listed.

Reply:
Done in 7c5300d3cab2e37c49ca1cdb800ef32c3b05d25b.

Reply:
I like your solution!

Just because I was curious, I ran that query anyways on `us.prairielearn.com`. Out of 35 distinct time zones, 14 are non-canonical.

```
America/Montreal
Asia/Harbin
Canada/Pacific
Europe/Amsterdam
Europe/Stockholm
Hongkong
Portugal
US/Arizona
US/Central
US/Eastern
US/Michigan
US/Mountain
US/Pacific
UTC
```

Reply:
Yep, great solution!